File: app/src/main/java/com/pluu/webtoon/ui/settting/LicenseActivity.java
package com.pluu.webtoon.ui.settting;

import android.net.Uri;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.customtabs.CustomTabsIntent;
import android.support.v4.content.ContextCompat;
import android.support.v7.app.ActionBar;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.MenuItem;

import com.pluu.event.RxBusProvider;
import com.pluu.webtoon.R;
import com.pluu.webtoon.adapter.LicenseAdapter;
import com.pluu.webtoon.event.RecyclerViewEvent;

import butterknife.Bind;
import butterknife.ButterKnife;
import rx.android.schedulers.AndroidSchedulers;
import rx.functions.Action1;
import rx.subscriptions.CompositeSubscription;

/**
 * License Activity
 * Created by PLUUSYSTEM-SURFACE on 2016-04-08.
 */
public class LicenseActivity extends AppCompatActivity {

    @Bind(R.id.listView)
    RecyclerView listView;

    private CompositeSubscription mCompositeSubscription;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_license);
        ButterKnife.bind(this);
        setupActionBar();
        initView();
    }

    /**
     * Set up the {@link android.app.ActionBar}, if the API is available.
     */
    private void setupActionBar() {
        ActionBar actionBar = getSupportActionBar();
        if (actionBar != null) {
            // Show the Up button in the action bar.
            actionBar.setDisplayHomeAsUpEnabled(true);
        }
    }

    private void initView() {
        listView.setLayoutManager(new LinearLayoutManager(this));
        listView.setAdapter(new LicenseAdapter(this));
    }

    @Override
    public void onResume() {
        super.onResume();
        mCompositeSubscription = new CompositeSubscription();
        mCompositeSubscription.add(
                RxBusProvider.getInstance()
                        .toObservable()
                        .observeOn(AndroidSchedulers.mainThread())
                        .subscribe(getBusEvent())
        );
    }

    @Override
    public void onPause() {
        super.onPause();
        mCompositeSubscription.unsubscribe();
    }

    @NonNull
    private Action1<Object> getBusEvent() {
        return new Action1<Object>() {
            @Override
            public void call(Object o) {
                if (o instanceof RecyclerViewEvent) {
                    itemClick((RecyclerViewEvent) o);
                }
            }
        };
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        int id = item.getItemId();
        if (id == android.R.id.home) {
            finish();
            return true;
        }
        return super.onOptionsItemSelected(item);
    }

    /**
     * RecyclerView Item Click
     * @param event Click Event
     */
    private void itemClick(RecyclerViewEvent event) {
        String url = getResources().getStringArray(R.array.license_url)[event.getPos()];

        // http://qiita.com/droibit/items/66704f96a602adec5a35

        final CustomTabsIntent tabsIntent = new CustomTabsIntent.Builder()
                .setShowTitle(true)
                .setToolbarColor(ContextCompat.getColor(this, R.color.theme_primary))
                .build();
        tabsIntent.launchUrl(this, Uri.parse(url));
    }

}

File: app/src/main/java/com/pluu/webtoon/ui/settting/LicenseActivity.kt
package com.pluu.webtoon.ui.settting

import android.net.Uri
import android.os.Bundle
import android.view.MenuItem
import androidx.appcompat.app.AppCompatActivity
import androidx.browser.customtabs.CustomTabsIntent
import androidx.core.content.ContextCompat
import androidx.lifecycle.lifecycleScope
import androidx.recyclerview.widget.LinearLayoutManager
import com.pluu.event.EventBus
import com.pluu.webtoon.R
import com.pluu.webtoon.adapter.LicenseAdapter
import com.pluu.webtoon.databinding.ActivityLicenseBinding
import com.pluu.webtoon.event.RecyclerViewEvent
import com.pluu.webtoon.utils.viewbinding.viewBinding
import kotlinx.coroutines.ExperimentalCoroutinesApi
import kotlinx.coroutines.FlowPreview
import kotlinx.coroutines.flow.collect

/**
 * License Activity
 * Created by pluu on 2017-05-05.
 */
class LicenseActivity : AppCompatActivity(R.layout.activity_license) {

    private val binding by viewBinding(ActivityLicenseBinding::bind)

    @FlowPreview
    @ExperimentalCoroutinesApi
    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        supportActionBar?.setDisplayHomeAsUpEnabled(true)
        initView()
    }

    @FlowPreview
    @ExperimentalCoroutinesApi
    private fun initView() {
        binding.listView.apply {
            layoutManager = LinearLayoutManager(this@LicenseActivity)
            adapter = LicenseAdapter(this@LicenseActivity)
        }

        lifecycleScope.launchWhenResumed {
            registerViewEvent()
        }
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if (item.itemId == android.R.id.home) {
            finish()
            return true
        }
        return super.onOptionsItemSelected(item)
    }

    @FlowPreview
    @ExperimentalCoroutinesApi
    private suspend fun registerViewEvent() {
        EventBus.subscribeToEvent<RecyclerViewEvent>()
            .collect {
                itemClick(it)
            }
    }

    /**
     * RecyclerView Item Click
     * @param event Click Event
     */
    private fun itemClick(event: RecyclerViewEvent) {
        val url = resources.getStringArray(R.array.license_url)[event.pos]

        // http://qiita.com/droibit/items/66704f96a602adec5a35

        CustomTabsIntent.Builder()
            .setShowTitle(true)
            .setToolbarColor(ContextCompat.getColor(this, R.color.theme_primary))
            .build()
            .launchUrl(this, Uri.parse(url))
    }
}
